src/wp-includes/rest-api/endpoints/class-wp-rest-block-types-controller.php

```diff
@@ -512,6 +512,59 @@ public function get_item_schema() {
 					'context'     => array( 'embed', 'view', 'edit' ),
 					'readonly'    => true,
 				),
+				'variations'       => array(
+					'description' => __( 'Block variations.' ),
+					'type'        => 'array',
+					'items'       => array(
+						'type'       => 'object',
+						'properties' => array(
+							'name'        => array(
+								'description' => __( 'The unique and machine-readable name.' ),
+								'type'        => 'string',
+								'required'    => true,
+							),
+							'title'       => array(
+								'description' => __( 'A human-readable variation title.' ),
+								'type'        => 'string',
+								'required'    => true,
+							),
+							'description' => array(
+								'description' => __( 'A detailed variation description.' ),
+								'type'        => 'string',
+								'required'    => false,
+							),
+							'category'    => array(
+								'description' => __( 'Block type category classification, used in search interfaces to arrange block types by category.' ),
+								'type'        => 'string',
+								'required'    => false,
+							),
+							'icon'        => array(
+								'description' => __( 'An icon helping to visualize the variation.' ),
+								'type'        => array( 'string', 'null' ),
+								'required'    => false,
+							),
+							'isDefault'   => array(
+								'description' => __( 'Indicates whether the current variation is the default one. Defaults to `false`' ),
+								'type'        => 'boolean',
+								'required'    => false,
+								'default'     => false,
+							),
+							'attributes'  => array(
+								'description'          => __( 'Block attributes.' ),
+								'type'                 => array( 'object', 'null' ),
+								'properties'           => array(),
+								'default'              => null,
+								'additionalProperties' => array(
+									'type' => 'object',
+								),
+							),
+							//TODO: innerBlocks, example, scope, keywords, isActive
```

Comment:
Note: `isActive` is a JavaScript function so it can't be encoded in PHP.

Reply:
For other reviewers, I ended up needing to add initialize this on the client in a `blocks.registerBlockType` filter https://github.com/WordPress/gutenberg/pull/29095/files#diff-2729a268ee263e041b843c649ade212b17133440dcf6ade630f1b7bee62db574R40